tests/scalers/temporal/temporal_test.go

```diff
@@ -0,0 +1,265 @@
+)
+
+var (
+	testNamespace = fmt.Sprintf("%s-ns", testName)
+
+	TemporalDeploymentName = fmt.Sprintf("temporal-%s", testName)
+
+	scaledObjectName = fmt.Sprintf("%s-so", testName)
+	deploymentName   = fmt.Sprintf("%s", testName)
+)
+
+type templateData struct {
+	DeploymentName         string
+	TestNamespace          string
+	TemporalDeploymentName string
+	ScaledObjectName       string
+}
+
+const (
+	temporalServiceTemplate = `
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{.TemporalDeploymentName}}
+  namespace: {{.TestNamespace}}
+spec:
+  type: ClusterIP
+  ports:
+    - port: 7233
+      protocol: TCP
+      targetPort: 7233
+  selector:
+    app: temporal
+`
+
+	temporalDeploymentTemplate = `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{.TemporalDeploymentName}}
+  namespace: {{.TestNamespace}}
+  labels:
+    app: temporal
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: temporal
+  template:
+    metadata:
+      labels:
+        app: temporal
+    spec:
+      containers:
+        - name: ubuntu
+          image: ubuntu:latest
+          command: ["bash", "-c"]
+          args:
+            - |
+              apt-get update && apt-get install -y curl && \
+              curl -sSf https://temporal.download/cli.sh | sh && \
+              PATH="\$PATH:/root/.temporalio/bin" >> ~/.bashrc && \
+              source ~/.bashrc && \
+              temporal server start-dev --namespace v2 --ip 0.0.0.0
+          ports:
+            - containerPort: 7233
+          livenessProbe:
+            exec:
+              command:
+                - /bin/sh
+                - -ec
+                - test $(ps -ef | grep -v grep | grep "temporal server" | wc -l) -eq 1
+            failureThreshold: 5
+            initialDelaySeconds: 10
+            periodSeconds: 30
+            successThreshold: 1
+            timeoutSeconds: 1
+          readinessProbe:
+            exec:
+              command:
+                - /bin/sh
+                - -ec
+                - test $(ps -ef | grep -v grep | grep "temporal server" | wc -l) -eq 1
+            failureThreshold: 5
+            initialDelaySeconds: 10
+            periodSeconds: 30
+            successThreshold: 1
+            timeoutSeconds: 1
+`
+
+	scaledObjectTemplate = `
+apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: {{.ScaledObjectName}}
+  namespace: {{.TestNamespace}}
+spec:
+  scaleTargetRef:
+    name: {{.DeploymentName}}
+  pollingInterval: 5
+  cooldownPeriod:  10
+  minReplicaCount: 0
+  maxReplicaCount: 1
+  advanced:
+    horizontalPodAutoscalerConfig:
+      behavior:
+        scaleDown:
+          stabilizationWindowSeconds: 10
+  triggers:
+  - type: temporal
+    metadata:
+      namespace: default
+      queueName: hello-task-queue
+      targetQueueSize: "2"
+      activationTargetQueueSize: "3"
+      endpoint: {{.TemporalDeploymentName}}.{{.TestNamespace}}.svc.cluster.local:7233
+`
+
+	deploymentTemplate = `
```

Comment:
Having an nginx deployment hooked up to a Temporal scaler is misleading. The Temporal worker should be in a deployment, and that is what should be being scaled.

Reply:
This might require a lot of work, as we would need to mimic long-running tasks and related scenarios. Do we really need this level of detail for the e2e tests?

Reply:
Done.